substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -37,11 +37,34 @@
 /// });
 /// ```
 #[must_use]
-pub struct StorageNoopGuard(sp_std::vec::Vec<u8>);
+pub struct StorageNoopGuard {
+	storage_root: sp_std::vec::Vec<u8>,
+	error_message: String,
```

Comment:
Ah damn, looks like this does not work since we dont have strings in the benchmarking no-std context 🤦   
Sorry, then it needs to be `&'static str`, or `&'a str` with the lifetime of the guard.

Reply:
haha OK I've reverted that commit and switched to the lifetime of the guard, I now see that since the `Drop` `impl` block has the same lifetime I didn't need static in the first place - I was thinking about it in the wrong way.